src/credentials/sharedCredentials.ts

```diff
@@ -0,0 +1,71 @@
+/*!
+ * Copyright 2020 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+import { join } from 'path'
+import { loadSharedConfigFiles, Profile, SharedConfigFiles } from '../shared/credentials/credentialsFile'
+import { EnvironmentVariables } from '../shared/environmentVariables'
+import { SystemUtilities } from '../shared/systemUtilities'
+import { union } from '../shared/utilities/collectionUtils'
+
+export function getCredentialsFilename(): string {
+    const env = process.env as EnvironmentVariables
+
+    return env.AWS_SHARED_CREDENTIALS_FILE || join(SystemUtilities.getHomeDirectory(), '.aws', 'credentials')
+}
+
+export function getConfigFilename(): string {
+    const env = process.env as EnvironmentVariables
+
+    return env.AWS_CONFIG_FILE || join(SystemUtilities.getHomeDirectory(), '.aws', 'config')
+}
+
+export async function loadSharedCredentialsProfiles(): Promise<Map<string, Profile>> {
+    const profiles = new Map<string, Profile>()
+    const profileData = await loadSharedConfigFiles()
+
+    const profileNames = getAllProfileNames(profileData)
+
+    for (const profileName of profileNames) {
+        const profile = mergeProfileProperties(
+            profileData.credentialsFile[profileName],
+            profileData.configFile[profileName]
+        )
+
+        profiles.set(profileName, profile)
+    }
+
+    return profiles
+}
+
+function getAllProfileNames(sharedCredentialsData: SharedConfigFiles): string[] {
+    const profileNames = union(
+        Object.keys(sharedCredentialsData.configFile),
+        Object.keys(sharedCredentialsData.credentialsFile)
+    )
+
+    return [...profileNames]
+}
+
+function mergeProfileProperties(credentialsProfile?: Profile, configProfile?: Profile): Profile {
+    let profile: Profile = {}
+
+    // Start with the config profile (if exists), then apply any credentials profile properties on top
+    if (configProfile) {
+        profile = configProfile
+    }
+
+    if (credentialsProfile) {
+        for (const index of Object.keys(credentialsProfile)) {
+            profile[index] = credentialsProfile[index]
+        }
+    }
+
+    return profile
+}
+
+export async function updateAwsSdkLoadConfigEnvironmentVariable(): Promise<void> {
```

Comment:
Unfortunately, we want this envvar to reflect reality as close to when we're using shared credentials as possible. Otherwise you can get into states where errors are raised due to missing files, or where profile data is ignored in newly created files.